charts/tenantfilesystem/templates/tenantfilesystem-composition.yaml

```diff
@@ -0,0 +1,854 @@
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            manifest:
+              apiVersion: storage.k8s.io/v1
+              kind: StorageClass
+              metadata:
+                name:
+                annotations:
+                  wadtfy.bmwgroup.net/restricted-namespaces: ""
+              mountOptions:
+              - tls
+              - iam
+              parameters:
+                az: ""
+                csi.storage.k8s.io/provisioner-secret-name: ""
+                csi.storage.k8s.io/provisioner-secret-namespace: ""
+                directoryPerms: "700"
+                fileSystemId: ""
+                provisioningMode: efs-ap
+                gidRangeStart: "50000"
+                gidRangeEnd: "7000000"
+                basePath: ""
+              provisioner: efs.csi.aws.com
+              reclaimPolicy: Delete
+              volumeBindingMode: Immediate
+          providerConfigRef:
+            name: provider-config-k8s
+      patches:
+        - type: PatchSet
+          patchSetName: "deletionPolicy"
+        - type: PatchSet
+          patchSetName: "metadata"
+        - fromFieldPath: "spec.storageClass.reclaimPolicy"
+          toFieldPath: "spec.forProvider.manifest.reclaimPolicy"
+        - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+          toFieldPath: "spec.forProvider.manifest.metadata.annotations[wadtfy.bmwgroup.net/restricted-namespaces]"
+        - fromFieldPath: "spec.region"
+          toFieldPath: "spec.forProvider.manifest.parameters.az"
+          policy:
+            fromFieldPath: Required
+        - fromFieldPath: "status.storageClassSecret"
+          toFieldPath: "spec.forProvider.manifest.parameters[csi.storage.k8s.io/provisioner-secret-name]"
+          policy:
+            fromFieldPath: Required
+        - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+          toFieldPath: "spec.forProvider.manifest.parameters[csi.storage.k8s.io/provisioner-secret-namespace]"
+        - fromFieldPath: "status.fileSystemID"
+          toFieldPath: "spec.forProvider.manifest.parameters.fileSystemId"
+          policy:
+            fromFieldPath: Required
+        - fromFieldPath: "spec.storageClass.uid"
+          toFieldPath: "spec.forProvider.manifest.parameters.uid"
+        - fromFieldPath: "spec.storageClass.gid"
+          toFieldPath: "spec.forProvider.manifest.parameters.gid"
+        - fromFieldPath: "spec.storageClass.gidRangeStart"
+          toFieldPath: "spec.forProvider.manifest.parameters.gidRangeStart"
+        - fromFieldPath: "spec.storageClass.gidRangeEnd"
+          toFieldPath: "spec.forProvider.manifest.parameters.gidRangeEnd"
+        - fromFieldPath: "spec.storageClass.basePath"
+          toFieldPath: "spec.forProvider.manifest.parameters.basePath"
+
+    - name: Role
```

Comment:
can you add a k8s prefix to kubernetes resources and an aws prefix to aws resources, please? otherwise this becomes quite confusing.
please do that for all resources, not just this one.